packages/core/src/ProxyConfiguration.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * Proxy server configuration.
+ */
+class ProxyConfiguration {
+    constructor(
+        /**
+         * Proxy type.
+         */
+        readonly type: ProxyType,
+        /**
+         * Proxy address. Can be either in the IP address format, ex. "1.1.1.1", or hostname, ex. "example.com".
+         */
+        readonly address: string,
+        /**
+         * Proxy port.
+         */
+        readonly port: number,
+        /**
+         * Username for Basic authentication scheme.
+         *
+         * Note: in case of Android, credentials are working out of the box only for "http" or "https" proxy types.
+         * In case of "socks" type you need to create a native module which will set "java.net.Authenticator", because
+         * OkHttp library doesn't support SOCKS+authentication proxy scheme.
+         */
+        readonly username?: string,
```

Comment:
is it possible to make add default null values for those ?

Reply:
why it is needed? if not defined, it will have `undefined` value.

Reply:
I was thinking not to have to call the constructor with `null` values or maybe it is already doing that ?

Reply:
It is optional, so the following call is valid `ProxyConfiguration(ProxyType.HTTP, "1.1.1.1", 8080)`. In this case `proxyConfiguration.username` will be just `undefined` and something like `if (proxyConfiguration.username)` will evaluate to `false`.